### PR TITLE
Improving ReadMe and adding comments to the code

### PR DESCRIPTION
**Pull request by jojosuelobo**
For the project's ReadMe, its structure was improved in a basic way, for a better visualization. In addition, some more comments were added to the code in the hope of improving the understanding of its logic for future uses and maintenance.

### CODE REVIEW

File: internal/errors/errors.go [hunk 1]
```diff
@@ -12,8 +12,8 @@ type SourcePosition struct {
 
 // ExtodanError represents a generic error in the Extodan language.
 type ExtodanError struct {
-    Message string
-    Position SourcePosition
+    Message   string         // A human-readable error message.
```
Comment (GarretTomlin) [inline internal/errors/errors.go:15]: the message is self documented

File: internal/errors/errors.go [hunk 2]
```diff
@@ -48,46 +48,44 @@ type SemanticError struct {
 // IOValidationError represents an error related to I/O validation.
 type IOValidationError struct {
     ExtodanError
-    Path string
+    Path string // The path associated with the I/O validation error.
```
Comment (GarretTomlin) [inline internal/errors/errors.go:51]: each individual variable don't need to be documented because the function itself is already documented

**Review by GarretTomlin — CHANGES_REQUESTED** [review]

**Comment by jojosuelobo** [comment]
The quoted tweaks have been changed to remove comments on self documented variables

**Comment by GarretTomlin** [comment]
@jojosuelobo  you broke something the test fail

**Pull request closed without merging**